### PR TITLE
Add support for pip 8.x

### LINKED ISSUE #112: pip v8 "TypeError: __init__() got an unexpected keyword argument 'allow_unverified'"
Updating to pip v8.0.0 gives me this TypeError with peep v2.5.0.
```
vagrant ~/treeherder $ ./bin/peep.py install -r requirements/common.txt -r requirements/dev.txt

Oh no! Peep had a problem while trying to do stuff. Please write up a bug report
with the specifics so we can fix it:

https://github.com/erikrose/peep/issues/new

Here are some particulars you can copy and paste into the bug report:

peep: (2, 5, 0)
python: '2.7.11 (default, Dec 15 2015, 16:48:05) \n[GCC 4.8.4]'
pip: '8.0.0'
Command line:  ['./bin/peep.py', 'install', '-r', 'requirements/common.txt', '-r', 'requirements/dev.txt']
Traceback (most recent call last):
  File "./bin/peep.py", line 956, in <module>
    exit(main())
  File "./bin/peep.py", line 928, in main
    return commands[argv[1]](argv[2:])
  File "./bin/peep.py", line 860, in peep_install
    for path in req_paths))
  File "./bin/peep.py", line 860, in <genexpr>
    for path in req_paths))
  File "./bin/peep.py", line 835, in downloaded_reqs_from_path
    finder = package_finder(argv)
  File "./bin/peep.py", line 386, in package_finder
    return PackageFinder(index_urls=index_urls, **kwargs)
TypeError: __init__() got an unexpected keyword argument 'allow_unverified'
```

I know hashing support has been added to pip v8, however if peep supported pip v8, it would make the transition easier, since we could update the pip used first, and only then update our requirements files/{pip,peep} invocations :-)

### PR DESCRIPTION
**Pull request by edmorley**
Pip 8 has deprecated the `--allow-external`, `--allow-unverified` and `--allow-all-external` options, since they they now a no-op after changes to PyPI, even with older versions of pip:
https://github.com/pypa/pip/pull/3070#issuecomment-137464224

Removing handling for these in peep does not affect older versions of pip, but avoids an exception for each removed option with pip v8:
`TypeError: __init__() got an unexpected keyword argument 'allow_unverified'`

Whilst pip v8 adds native hashing support, it's helpful for peep to support v8 so that users can separate the "update pip" and "switch from peep to pip hashes" steps.

Fixes #112.

CC @erikrose

### CODE REVIEW

**Comment by erikrose** [comment]
You're a really good person, @edmorley. :-)

File: peep.py [hunk 1]
```diff
@@ -362,9 +362,11 @@ def package_finder(argv):
     # Carry over PackageFinder kwargs that have [about] the same names as
     # options attr names:
     possible_options = [
-        'find_links', FORMAT_CONTROL_ARG, 'allow_external', 'allow_unverified',
```
Comment (erikrose) [inline peep.py]: I don't think we ought to remove allow_external and allow_all_external; there's nothing wrong (from a security standpoint—especially when checking hashes) with downloading from sites other than PyPI. Can we stick these in a branch? We might as well bring back allow_unverified while we're at it.

Comment (edmorley) [inline peep.py]: They are a no-op now, even on older pip versions, unless I've misunderstood? (see commit message)

**Comment by edmorley** [comment]
I have a fix for the test failures locally (unpinned virtualenv in the travis tox install ironically :-)).

File: peep.py (hunk 1 above)
Comment (erikrose) [inline peep.py]: Oh! Right you are. I suppose people running their own indices could still be using the old protocol version, but they'd probably also have to be using an older version of pip, so they'd be welcome to continue using an old version of peep as well. Carry on.

**Comment by erikrose** [comment]
You sure figured that out fast!

File: .travis.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ sudo: false
 language: python
 install:
   # Stick to a known-good version of tox, just in case:
-  - pip install tox==1.9.2 flake8==2.4.0
+  - pip install tox==1.9.2 virtualenv==14.0.0 flake8==2.4.0
```
Comment (erikrose) [inline .travis.yml]: Good call. I was just about to suggest this when your new commits landed. :-)

**Comment by edmorley** [comment]
Eugh there were failures with virtualenv==14.0.0 after all; I've just gone full circle again (after a few bouts of indecision) and pinned it to the previously working version. Serves me right for being lazy and using Travis to test instead of doing so locally (I don't have most of the interpreters installed) :-)

**Comment by erikrose** [comment]
wfm. FWIW, I also don't care if we continue to support Python 3.2; pip just dropped support, and I don't imagine it has much userbase.

**Pull request merged**